### PR TITLE
ref: Remove now unused discover query code

### PR DESCRIPTION
**Pull request by markstory**
Remove the automatic dataset detection logic from sentry. We have switched over to the snuba based implementation of this. Not having dataset selection in sentry also means we can remove some (but not all) of the supporting logic as well.

### CODE REVIEW

**Review by iProgramStuff — APPROVED** [review]
Looks good to me. I don't see this affecting any of my work on the groups dataset / issue search backend.

Looks a lot cleaner now not passing the dataset around....and that detect_dataset method was a monster.

File: src/sentry/api/bases/organization_events.py [hunk 1]
```diff
@@ -70,12 +68,4 @@ def get_snuba_query_args_legacy(self, request, organization):
...
         }
 
-        # 'legacy' endpoints cannot access transactions dataset.
-        # as they often have assumptions about which columns are returned.
-        dataset = snuba.detect_dataset(snuba_args)
-        if dataset != Dataset.Events:
-            raise OrganizationEventsError(
-                "Invalid query. You cannot reference non-events data in this endpoint."
-            )
-
```
Comment (iProgramStuff) [inline src/sentry/api/bases/organization_events.py]: What are we doing to make sure they aren't referencing transaction data now?

Comment (iProgramStuff) [reply src/sentry/api/bases/organization_events.py]: I assume this was returning the transactions dataset before, but now snuba will determine dataset - so are we checking somewhere else that they won't be needing transaction data?

Comment (markstory) [reply src/sentry/api/bases/organization_events.py]: Previously queries to this endpoint would end up in `dataset_query()` and have the dataset automatically detected. Now that `transform_aliases_and_query` is pinned to the events dataset this check is redundant.

**Review by wedamija — APPROVED** [review]
Nice, love to delete code

**Pull request merged**